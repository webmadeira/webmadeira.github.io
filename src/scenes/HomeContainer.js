import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import Root from './components/Root/Root'
import Header from './components/Header/Header'
import Body from './components/Body/Body'
import Footer from './components/Footer/Footer'
import EventDescription from './components/EventDescription/EventDescription'
import Talks from './components/Talks/Talks'
import { getOrganizationDetails } from '../store/actions/organization'
import { getEventDetails } from '../store/actions/event'
import { getSpeakers } from '../store/actions/speaker'
import Speakers from './components/Speakers/Speakers'

class HomeContainer extends React.Component {
  static propTypes = {
    getOrganizationDetails: PropTypes.func.isRequired,
    getEventDetails: PropTypes.func.isRequired,
    getSpeakers: PropTypes.func.isRequired,
    event: PropTypes.shape({
      title: PropTypes.string.isRequired,
      date: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.instanceOf(Date),
      ]),
      description: PropTypes.string.isRequired,
      talks: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
    }).isRequired,
    speakers: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
    organization: PropTypes.shape({
      logo: PropTypes.string.isRequired,
    }).isRequired,
  }

  componentDidMount() {
    this.props.getOrganizationDetails()
    this.props.getEventDetails(0)
    this.props.getSpeakers(2018)
  }

  render() {
    const { speakers } = this.props
    const { logo } = this.props.organization
    const {
      title,
      date,
      description,
      talks,
    } = this.props.event

    return (
      <Root>
        <Header title={title} date={date} logo={logo} />
        <Body>
          <EventDescription
            description={description}
          >
            <Talks numTalks={talks.length} />
          </EventDescription>
          <Speakers speakers={speakers} />
        </Body>
        <Footer />
      </Root>
    )
  }
}

function mapStateToProps({ event, organization, speaker: { speakers } }) {
  let newDate = event.date
  let modifinedEvent = event

  if ((event.date) && (typeof event.date === 'object')) {
    const month = newDate.toLocaleString('en-us', { month: 'short' })
    const day = event.date.getDate()

    newDate = `<${month}.${day} />`
    modifinedEvent = {
      ...event,
      date: newDate,
    }
  }

  return {
    speakers,
    organization,
    event: modifinedEvent,
  }
}

const mapDispatchToProps = {
  getOrganizationDetails,
  getEventDetails,
  getSpeakers,
}

export default connect(mapStateToProps, mapDispatchToProps)(HomeContainer)
