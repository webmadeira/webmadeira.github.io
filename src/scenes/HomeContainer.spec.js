import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'
import HomeContainer from './HomeContainer'

const middlewares = []
const mockStore = configureStore(middlewares)

describe('HomeContainer component', () => {
  let mocks

  beforeEach(() => {
    mocks = {
      store: {
        event: {
          title: 'Test title',
          date: 'Test date',
          description: 'Test description',
          talks: [
            { id: 0, description: 'Test talk 1' },
            { id: 1, description: 'Test talk 2' },
          ],
        },
        organization: {
          logo: 'logo path',
        },
        speaker: {
          speakers: [
            {
              id: 0,
              name: 'André Gonçalves',
              company: 'Freelancer',
              jobTitle: 'Web Dev Consultant',
              personalWebsite: 'https://github.com/andregoncalves',
              bwPhoto: 'https://avatars3.githubusercontent.com/u/1074?s=460&v=4',
              photo: 'https://avatars3.githubusercontent.com/u/1074?s=460&v=4',
            },
            {
              id: 1,
              name: 'Aurélio Pita',
              company: 'Blip',
              jobTitle: 'Front End Developer',
              jobWebsite: 'https://blip.pt/',
              personalWebsite: 'https://github.com/jaapaurelio',
              bwPhoto: 'https://avatars3.githubusercontent.com/u/2837728?s=460&v=4',
              photo: 'https://avatars3.githubusercontent.com/u/2837728?s=460&v=4',
            },
          ],
        },
      },
    }
  })

  it('must render properly', () => {
    const store = mockStore(mocks.store)
    const component = shallow(<HomeContainer store={store} />)

    expect(component).toMatchSnapshot()
  })

  it('must render properly with date', () => {
    const store = mockStore({
      ...mocks.store,
      event: {
        ...mocks.store.event,
        date: new Date('2018-08-10'),
      },
    })

    const component = shallow(<HomeContainer store={store} />)

    expect(component).toMatchSnapshot()
  })
})
