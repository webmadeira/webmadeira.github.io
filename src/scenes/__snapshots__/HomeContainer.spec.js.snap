// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(HomeContainer)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "event": Object {
        "date": "Test date",
        "description": "Test description",
        "talks": Array [
          Object {
            "description": "Test talk 1",
            "id": 0,
          },
          Object {
            "description": "Test talk 2",
            "id": 1,
          },
        ],
        "title": "Test title",
      },
      "getEventDetails": [Function],
      "getOrganizationDetails": [Function],
      "getSpeakers": [Function],
      "organization": Object {
        "logo": "logo path",
      },
      "speakers": Array [
        Object {
          "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
          "company": "Freelancer",
          "id": 0,
          "jobTitle": "Web Dev Consultant",
          "name": "André Gonçalves",
          "personalWebsite": "https://github.com/andregoncalves",
          "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
        },
        Object {
          "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
          "company": "Blip",
          "id": 1,
          "jobTitle": "Front End Developer",
          "jobWebsite": "https://blip.pt/",
          "name": "Aurélio Pita",
          "personalWebsite": "https://github.com/jaapaurelio",
          "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
        },
      ],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "event": Object {
          "date": "Test date",
          "description": "Test description",
          "talks": Array [
            Object {
              "description": "Test talk 1",
              "id": 0,
            },
            Object {
              "description": "Test talk 2",
              "id": 1,
            },
          ],
          "title": "Test title",
        },
        "getEventDetails": [Function],
        "getOrganizationDetails": [Function],
        "getSpeakers": [Function],
        "organization": Object {
          "logo": "logo path",
        },
        "speakers": Array [
          Object {
            "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
            "company": "Freelancer",
            "id": 0,
            "jobTitle": "Web Dev Consultant",
            "name": "André Gonçalves",
            "personalWebsite": "https://github.com/andregoncalves",
            "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
          },
          Object {
            "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
            "company": "Blip",
            "id": 1,
            "jobTitle": "Front End Developer",
            "jobWebsite": "https://blip.pt/",
            "name": "Aurélio Pita",
            "personalWebsite": "https://github.com/jaapaurelio",
            "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
          },
        ],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HomeContainer component must render properly with date 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(HomeContainer)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "event": Object {
        "date": "<Aug.10 />",
        "description": "Test description",
        "talks": Array [
          Object {
            "description": "Test talk 1",
            "id": 0,
          },
          Object {
            "description": "Test talk 2",
            "id": 1,
          },
        ],
        "title": "Test title",
      },
      "getEventDetails": [Function],
      "getOrganizationDetails": [Function],
      "getSpeakers": [Function],
      "organization": Object {
        "logo": "logo path",
      },
      "speakers": Array [
        Object {
          "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
          "company": "Freelancer",
          "id": 0,
          "jobTitle": "Web Dev Consultant",
          "name": "André Gonçalves",
          "personalWebsite": "https://github.com/andregoncalves",
          "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
        },
        Object {
          "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
          "company": "Blip",
          "id": 1,
          "jobTitle": "Front End Developer",
          "jobWebsite": "https://blip.pt/",
          "name": "Aurélio Pita",
          "personalWebsite": "https://github.com/jaapaurelio",
          "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
        },
      ],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "event": Object {
          "date": "<Aug.10 />",
          "description": "Test description",
          "talks": Array [
            Object {
              "description": "Test talk 1",
              "id": 0,
            },
            Object {
              "description": "Test talk 2",
              "id": 1,
            },
          ],
          "title": "Test title",
        },
        "getEventDetails": [Function],
        "getOrganizationDetails": [Function],
        "getSpeakers": [Function],
        "organization": Object {
          "logo": "logo path",
        },
        "speakers": Array [
          Object {
            "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
            "company": "Freelancer",
            "id": 0,
            "jobTitle": "Web Dev Consultant",
            "name": "André Gonçalves",
            "personalWebsite": "https://github.com/andregoncalves",
            "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
          },
          Object {
            "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
            "company": "Blip",
            "id": 1,
            "jobTitle": "Front End Developer",
            "jobWebsite": "https://blip.pt/",
            "name": "Aurélio Pita",
            "personalWebsite": "https://github.com/jaapaurelio",
            "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
          },
        ],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
